libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManagerInternal.swift

```diff
@@ -210,5 +213,166 @@ public class NativeLoginManagerInternal: NSObject, NativeLoginManager {
         let hash = SHA256.hash(data: data)
         return urlSafeBase64Encode(data: hash.dataRepresentation)
     }
+    
+    // MARK: Headerless, Password-Less Login Via One-Time-Passcode
+    
+    /// The most recently obtained OTP identifier in the headerless, password-less login flow.
+    private var otpIdentifier: String?
+    
+    public func submitOtpRequest(
+        username: String,
+        reCaptchaToken: String,
+        reCaptchaSiteKeyId: String?,
+        googleCloudProjectId: String?,
+        isReCaptchaEnterprise: Bool,
+        otpVerificationMethod: OtpVerificationMethod) async throws -> NativeLoginResult
+    {
+        
+        // Validate parameters.
+        if !isValidUsername(
+            username: username.trimmingCharacters(in: .whitespacesAndNewlines)
+        ) {
+            return .invalidUsername
+        }
+        
+        /*
+         * Create the OTP request body with the provided parameters. Note: The
+         * `emailtemplate` parameter isn't supported here, but could be added in
+         * the future.
+         */
+        // Determine the reCAPTCHA parameter for non-enterprise reCAPTCHA
+        let reCaptchaParameter: String? = if (isReCaptchaEnterprise) {
+            nil
+        } else {
+            reCaptchaToken
+        }
+        // Determine the reCAPTCHA "event" parameter for enterprise reCAPTCHA
+        let reCaptchaEventParameter: OtpRequestBodyReCaptchaEvent? = if (isReCaptchaEnterprise) {
+            OtpRequestBodyReCaptchaEvent(
+                token: reCaptchaToken,
+                siteKey: try reCaptchaSiteKeyId ?? { throw NSError(
+                    domain:NativeLoginManagerInternal.errorDomain,
+                    code:-1,
+                    userInfo: [NSLocalizedDescriptionKey: "A reCAPTCHA site key wasn't and must be provided when using enterprise reCAPATCHA."])}(),
+                projectId: try googleCloudProjectId ?? { throw NSError(
+                    domain:NativeLoginManagerInternal.errorDomain,
+                    code:-1,
+                    userInfo: [NSLocalizedDescriptionKey: "A Google Cloud project id wasn't and must be provided when using enterprise reCAPATCHA."])}()
+            )
+        } else {
+            nil
+        }
+        // Determine the OTP verification method.
+        let otpVerificationMethodString = switch (otpVerificationMethod) {
+        case .email: "email"
+        case .sms: "sms"
+        }
+        // Generate the OTP request body.
+        let requestBodyString = try {
+            do { return String(
+                data: try JSONEncoder().encode(
+                    OtpRequestBody(
+                        recaptcha: reCaptchaParameter,
+                        recaptchaevent: reCaptchaEventParameter,
+                        username: username,
```

Comment:
Does the server trim whitespace and new lines?  I guess I never checked.

Reply:
I confirmed the service is _not_ trimming white space for these, alas.

Reply:
So should we be using the trimmed version here?